portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,42 +34,45 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> comments;
-
-  @Override
-  public void init() {
-    comments = new ArrayList<String>();
-    comments.add("Wow, Obi nice website!");
-    comments.add("I love the color scheme");
-    comments.add("We love you Obi!");
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(comments);
-    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> commentList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, text, timestamp);
+      commentList.add(comment);
+    }
+
+    Gson gson = new Gson();
+
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(commentList));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
```

Comment:
This comment is really vague, so it doesn't help me understand the code more. It raises more questions than it answers. What is the input and where is it sent from? What/where is the form?  Either make this comment more specific or delete it. I can get a pretty good feel for what the input is just by reading the code and seeing that you grab the text from the `"comment-input"` parameter, so in this case the comment is probably redundant.

In general avoid using vague terms like "input". Use more specific terms like "comment text" or "timestamp".